src/scripts/main.js

```diff
@@ -1 +1,61 @@
 'use strict';
+
+const showMessage = (message, className) => {
+  const div = document.createElement('div');
+
+  div.dataset.qa = 'notification';
+  div.classList.add(className);
+  div.textContent = message;
+  document.body.append(div);
+};
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+firstPromise.then(res => showMessage(res, 'success'))
+  .catch(res => showMessage(res, 'warning'));
+
+const secondPromise = new Promise(resolve => {
```

Comment:
1) I would recommend to move your resolvers into own functions
```
const clickPromise = new Promise(resolve => {
  document.addEventListener('click', () => {
    resolve();
  });
});

const contextMenuPromise = new Promise(resolve => {
  document.addEventListener('contextmenu', () => {
    resolve();
  });
});
```

2) apply Promise.race to the secondPromise
```
const secondPromise = new Promise((resolve, reject) => {
  Promise.race([clickPromise, contextMenuPromise])
    .then((value) => resolve('Second promise was resolved'));
});
```

3) apply Promise.all to the thirdPromise
```
const thirdPromise = new Promise(resolve => {
  Promise.all([clickPromise, contextMenuPromise])
    .then((value) => resolve('Third promise was resolved'));
});
```

Reply:
https://github.com/mate-academy/js_promises_practice_DOM/pull/485#discussion_r1170336542
check the previous comment, if you have some questions, just feel free to ask in `fe_chat`